aries_cloudagent/protocols/present_proof/v2_0/formats/handler.py

```diff
@@ -0,0 +1,127 @@
+
+        Returns:
+            The profile instance for this presentation exchange format
+
+        """
+        return self._profile
+
+    def get_format_identifier(self, message_type: str) -> str:
+        """Get attachment format identifier for format and message combination.
+
+        Args:
+            message_type (str): Message type for which to return the format identifier
+
+        Returns:
+            str: Issue credential attachment format identifier
+
+        """
+        return ATTACHMENT_FORMAT[message_type][self.format.api]
+
+    def get_format_data(self, message_type: str, data: dict) -> PresFormatAttachment:
+        """Get presentation format and attach objects for use in pres_ex messages."""
+        if self.format.api == "dif":
+            attach = AttachDecorator.data_json(data, ident=self.format.api)
+        elif self.format.api == "indy":
+            attach = AttachDecorator.data_base64(data, ident=self.format.api)
+
+        return (
+            V20PresFormat(
+                attach_id=self.format.api,
+                format_=self.get_format_identifier(message_type),
+            ),
+            attach,
+        )
+
+    @abstractclassmethod
+    def validate_fields(cls, message_type: str, attachment_data: dict) -> None:
+        """Validate attachment data for specific message type and format."""
+
+    @abstractmethod
+    async def create_bound_request(
+        self,
+        pres_ex_record: V20PresExRecord,
+        request_data: dict = None,
+    ) -> PresFormatAttachment:
+        """Create a presentation request bound to a proposal."""
+
+    @abstractmethod
+    async def create_pres(
+        self,
+        pres_ex_record: V20PresExRecord,
+        request_data: dict = None,
+    ) -> PresFormatAttachment:
+        """Create a presentation."""
+
+    @abstractmethod
+    async def receive_pres(
+        self, message: V20Pres, pres_ex_record: V20PresExRecord
+    ) -> None:
+        """Receive a presentation, from message in context on manager creation."""
+
+    @abstractmethod
+    async def verify_pres(self, pres_ex_record: V20PresExRecord) -> V20PresExRecord:
+        """Verify a presentation."""
+
+    @abstractmethod
+    async def create_exchange_for_proposal(
+        self,
+        pres_ex_record: V20PresExRecord,
+        pres_proposal_message: V20PresProposal,
+    ) -> None:
+        """Create a presentation exchange record for input presentation proposal."""
+
+    @abstractmethod
+    async def receive_pres_proposal(
+        self,
+        pres_ex_record: V20PresExRecord,
+        message: V20PresProposal,
+    ) -> None:
+        """Receive a presentation proposal from message in context on manager creation."""
+
+    @abstractmethod
+    async def create_exchange_for_request(
+        self,
+        pres_ex_record: V20PresExRecord,
+        pres_request_message: V20PresRequest,
+    ) -> None:
+        """Create a presentation exchange record for input presentation request."""
```

Comment:
The idea of the format handlers was that it would handle all the format specific stuff and let the manager handle the general protocol stuff such as creating the exchange record. Is there a specific requirement to let the handler create the exchange records?

Reply:
Updated in create_exchange_for_proposal, receive_pres_proposal & create_exchange_for_request in  ../present_proof/v2_0/manager.py